### PR TITLE
test: team fetch/sync

### PR DESCRIPTION
**Pull request by alexandreferris**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

[AR-1221](https://wearezeta.atlassian.net/browse/AR-1221)

Added tests for:
- TeamMapper
- TeamRepositoryTest
- TeamDAO

### Dependencies (Optional)

Needs releases with:

- [X] #254 
- [X] #257

### CODE REVIEW

**Review by MohamadJaara — APPROVED** [review]
lgtm 🚀

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/team/TeamRepositoryTest.kt [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    fun setUp() {
+        teamRepository = TeamDataSource(
+            teamDAO = teamDAO,
+            teamMapper = teamMapper,
+            teamsApi = teamsApi
+        )
+    }
+
+    @Test
+    fun givenSelfUserExists_whenGettingTeamInfo_thenTeamInfoShouldBeReturned() = runTest {
```
Comment (vitorhugods) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/data/team/TeamRepositoryTest.kt:42]: We could add another test and assert if `teamDAO` was called to insert the mapped team info.

Something like `givenTeamApiReturnsSuccessfully_whenGettingTeamInfo_thenTeamInfoIsPersisted`.

Also, another couple of tests to make sure that errors in the API or DAO are propagated correctly would be nice:

* `givenTeamApiFails_whenGettingTeamInfo_thenTheFailureIsPropagated`
* `givenTeamDAOFails_whenGettingTeamInfo_thenTheFailureIsPropagated`

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/team/TeamRepositoryTest.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        )
+
+        given(teamMapper)
+            .function(teamMapper::fromApiModelToDaoModel)
+            .whenInvokedWith(oneOf(team))
+            .then { teamEntity }
+
+        given(teamDAO)
+            .coroutine { insertTeam(team = teamEntity) }
+            .then { } // returns Unit
```
Comment (vitorhugods) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/data/team/TeamRepositoryTest.kt:75]: You can initialise `teamDAO` like this, so you don't need to stub methods that return `Unit`:

```kotlin
    @Mock
    private val teamDAO = configure(mock(classOf<TeamDAO>())){
        stubsUnitByDefault = true
    }
```

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/team/TeamRepositoryTest.kt [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+        val result = teamRepository.fetchTeamById(teamId = "teamId")
+        assertEquals(Either.Right(Unit), result)
```
Comment (vitorhugods) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/data/team/TeamRepositoryTest.kt:78]: This tests only asserts success in the end, so we might as well just assert the success.

```suggestion
        result.shouldSucceed()
```

This way, if we ever change what `fetchTeamById` returns, maybe from `Either<Failure, Unit>` to `Either<Failure, TeamStuff>`, this test won't break unnecessarily.

**Review by vitorhugods — CHANGES_REQUESTED** [review]
Nicely done. But it would be cool to have a couple more tests here or there.

**Pull request merged**